tests/unit/lib/WUI/nhttp/server_tests.cpp

```diff
@@ -217,29 +217,22 @@ class MockServerDefs final : public ServerDefs {
     friend class MockServer;
     vector<shared_ptr<ConnInfo>> &infos;
     static const constexpr Selector *selectors_array[] = { &validate_request, &fake_index, &fake_api, &unknown_request };
-    altcp_pcb *mock_alloc_inner() {
+    altcp_pcb *new_conn() const {
         MockConn *conn = new MockConn;
         infos.push_back(conn->info);
         return conn;
     }
-    static altcp_pcb *mock_alloc(void *arg, uint8_t ip_type) {
-        return static_cast<MockServerDefs *>(arg)->mock_alloc_inner();
-    }
     string api_key;
 
 public:
     MockServerDefs(vector<shared_ptr<ConnInfo>> &conn_infos)
         : infos(conn_infos) {}
     virtual const Selector *const *selectors() const override { return selectors_array; }
     virtual const char *get_api_key() const override { return api_key.c_str(); }
-    virtual altcp_allocator_t listener_alloc() const override {
-        altcp_allocator_t alloc = {
-            mock_alloc,
-            (void *)this,
-        };
-        return alloc;
-    }
-    virtual uint16_t port() const override { return 80; }
+    virtual altcp_pcb *listener_alloc() const override {
+        auto conn = new_conn();
+        return altcp_listen(conn);
```

Comment:
```suggestion
        return altcp_listen(new_conn());
```

Reply:
I sometimes prefer not to nest the function calls too much, mostly for readability. In here, I split it mostly because I want to hint that creating the connection and listening are two separate actions, not two parts of a single one. Therefore, I'll prefer keeping it this way as it is.